doc/CMakeLists.txt

```diff
@@ -65,217 +60,387 @@ set(I18NSPHINXOPTS  ${SPHINXOPTS})
 
 set(DOXYFILE_IN ${CMAKE_CURRENT_LIST_DIR}/zephyr.doxyfile.in)
 set(DOXYFILE_OUT ${CMAKE_CURRENT_BINARY_DIR}/zephyr.doxyfile)
-set(RST_OUT ${CMAKE_CURRENT_BINARY_DIR}/rst)
 set(DOC_LOG ${CMAKE_CURRENT_BINARY_DIR}/doc.log)
 set(DOXY_LOG ${CMAKE_CURRENT_BINARY_DIR}/doxy.log)
 set(SPHINX_LOG ${CMAKE_CURRENT_BINARY_DIR}/sphinx.log)
 set(DOC_WARN ${CMAKE_CURRENT_BINARY_DIR}/doc.warnings)
-set(CONTENT_OUTPUTS ${CMAKE_CURRENT_BINARY_DIR}/extracted-content.txt)
 
 configure_file(${DOXYFILE_IN} ${DOXYFILE_OUT} @ONLY)
 
-# This command is used to copy all documentation source files into the build
-# directory,
 #
-# We need to make copies because Sphinx requires a single
-# documentation root directory, but Zephyr's documentation is
-# scattered around the tree in samples/, boards/, and doc/. Putting
-# them into a single rooted tree in the build directory is a
-# workaround for this limitation.
-set(EXTRACT_CONTENT_COMMAND
-  ${CMAKE_COMMAND} -E env
-  ${PYTHON_EXECUTABLE} scripts/extract_content.py
-  # Ignore any files in the output directory.
-  --ignore ${CMAKE_CURRENT_BINARY_DIR}
-  # Copy all files in doc to the rst folder.
-  "*:doc:${RST_OUT}"
-  # We want to copy the .rst files in samples/ and boards/ to the rst
-  # folder, and also the doc folder inside rst.
-  #
-  # Some files refer to items in samples/ and boards/ relative to
-  # their actual position in the Zephyr tree. For example, in
-  # subsystems/sensor.rst:
-  #
-  # .. literalinclude:: ../../samples/sensor/mcp9808/src/main.c
+# Macro to generate target for document content
+#
+# Note: this is a macro because variables set by this
+#       are being used in constructing other targets.
+#
+macro(construct_content_target scope)
+  if("${scope}" STREQUAL "")
```

Comment:
Please remove everything with `CONTENT_OUTPUTS` and the targets that depend on it, such as `sphinx-html` and `sphinx-latex`. Those are no longer valid since we simplified the build system, and are a leftover that should've been removed in the first place.

Reply:
This is because we no longer generate the list of `.rst` files since that was simply not effective